test/integration/loader.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import re
+import sys
+
+re_empty = re.compile("^$")
+re_req = re.compile("^>>> (.+)$")
+re_rsp = re.compile("^<<< (.+)$")
+re_stat = re.compile("^\+\+\+ (.+)$")
+
+
+def split_metrics(line):
+    """split a line of metric deltas into a dict, for example:
+
+    'get +1, get_hit +1' -> {'get':1, 'get_hit':1}
+    'request_free -1, request_parse +1' -> {'request_free':-1, 'request_parse':1}
+    """
+    metrics = line.split(',')
+    d = {}
+    for m in metrics:
+        name,delta = m.strip().split(' ')
+        d[name.strip()] = int(delta)
+
+    return d
+
+
+def load_seq(fname):
+    """Load the test (a sequence of commands and asserts) from a file.
+    Each command contains one or more lines of request, leading with '>>> ', and
+    one or more lines of response, leading with '<<< '. Commands are separated
+    by an empty line.
+    """
+    lines = open(fname).readlines()
+    if not re_empty.match(lines[-1]):  # ensure an empty line at the end
+      lines.append('\n')
+
+    seq = []
```

Comment:
You have a valid point about observing stats change mid-request though, which is something that we may want to do, and my assumption would force such sequence to be carried out by pretending we are sending two separate requests, which is not quite right.

I agree you suggestion is better in general, I would only add that blank lines can be added anytime to break up the flow visually, and can be ignored by the loader.